platform/src/main/java/org/stellar/anchor/platform/paymentobserver/StellarPaymentObserver.java

```diff
@@ -156,34 +171,60 @@ public StellarPaymentObserver build() {
     }
   }
 
-  public static void main(String[] args) throws InterruptedException {
-    KeyPair account1 =
-        KeyPair.fromSecretSeed("SCBYEX2YH7BH5WVKVR22RW2M3QQR3P2P3NLWIVNNNEZHJ3KZ52E2QKZN");
-    KeyPair account2 =
-        KeyPair.fromSecretSeed("SDPJLASIYSGX7ZKOJZIGJGYGC5F6XKHTPEA7NR2Y6YKKCHIBR2GAPCEZ");
+  @Override
+  public String getName() {
+    return "stellar_payment_observer";
+  }
 
-    PaymentListener dummyObserver =
-        new PaymentListener() {
-          @Override
-          public void onReceived(ObservedPayment payment) {
-            System.out.println("Received:" + new Gson().toJson(payment));
-          }
+  @Override
+  public List<String> getTags() {
+    return List.of("all", "event");
+  }
 
-          @Override
-          public void onSent(ObservedPayment payment) {
-            System.out.println("Sent:" + new Gson().toJson(payment));
-          }
-        };
-
-    StellarPaymentObserver watcher =
-        builder()
-            .horizonServer("https://horizon-testnet.stellar.org")
-            .accounts(List.of(account1.getAccountId(), account2.getAccountId()))
-            .observers(List.of(dummyObserver))
-            .build();
-
-    watcher.start();
-    Thread.sleep(300000);
-    watcher.shutdown();
+  @Override
+  public HealthCheckResult check(HealthCheckContext context) {
+    List<StreamHealth> results = new ArrayList<>();
+    for (SSEStream<OperationResponse> stream : streams) {
+      StreamHealth.StreamHealthBuilder builder = StreamHealth.builder();
+      builder.account(mapStreamToAccount.get(stream));
+      // populate executorService information
+      ExecutorService executorService = getField(stream, "executorService", null);
+      if (executorService != null) {
+        builder.threadShutdown(executorService.isShutdown());
+        builder.threadTerminated(executorService.isTerminated());
+      }
+
+      builder.stopped(getField(stream, "isStopped", new AtomicBoolean(false)).get());
+
+      AtomicReference<String> lastEventId = getField(stream, "lastEventId", null);
+      if (lastEventId != null) {
+        builder.lastEventId(lastEventId.get());
+      }
+
+      results.add(builder.build());
+    }
+
+    return new SPOHealthCheckResult(getName(), results);
```

Comment:
I think it'd be more efficient to use the `onFailure` and `onEvent` (with its try/catches) to periodically update a static variable.

Despite providing extra information, accessing the private variables seems too hacky.

Reply:
One of important check is to get the status of the SSEStream. It is possible that the thread died and the process is still running. 

I wish those variables are accessible from the class. However, the SDK was not designed to provide the status. That's why I added a function to access those variables. 

I had added an [issue #432 ](https://github.com/stellar/java-stellar-sdk/issues/432)in the Stellar Java SDK to make those variables non-private so that we may avoid the change of accessibility.